unstructured/ingest/cli/cmds/sharepoint.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+import click
+
+from unstructured.ingest.cli.common import (
+    add_recursive_option,
+    add_shared_options,
+    log_options,
+    map_to_processor_config,
+    map_to_standard_config,
+    run_init_checks,
+)
+from unstructured.ingest.logger import ingest_log_streaming_init, logger
+from unstructured.ingest.runner import sharepoint as sharepoint_fn
+
+
+@click.command()
+@click.option(
+    "--client-id",
+    default=None,
+    help="Sharepoint app client ID",
+)
+@click.option(
+    "--client-cred",
+    default=None,
+    help="Sharepoint app secret",
+)
+@click.option(
+    "--site",
+    default=None,
+    help="Sharepoint site url. Process either base url e.g https://[tenant].sharepoint.com \
+        or relative sites https://[tenant].sharepoint.com/sites/<site_name>.",
+)
+@click.option(
+    "--path",
+    default="Shared Documents",
+    help="Path from which to start parsing files. If the connector is to process all sites \
+    within the tenant this filter will be applied to all sites document libraries. \
+    Default 'Shared Documents'",
+)
+@click.option(
+    "--process-all",
+    is_flag=True,
+    default=False,
+    help="Process all sites within the tenant. Requires the app to be registered at a tenant \
+        level and a site url in https://[tenant]-admin.sharepoint.com.",
+)
```

Comment:
rather than having two parameters how about we process all sites if you don't specify a specific site with the --site flag?

also please call out this logic in the cli help text and the example

Reply:
no, too finicky and hard to discover. 
by "prefix is not present" do you mean if the --site option isn't set? hmm, what are we _currently_ doing if it --site flag is not?

Reply:
the site url would only have that address if we wanted to process all sites though, correct? if so, can we just detect if the --site url has that form and if so then process all sites?